packages/docgen/src/index.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+const { last } = require( 'lodash' );
+
+/**
+ * Internal dependencies
+ */
+const engine = require( './engine' );
+const defaultMarkdownFormatter = require( './markdown' );
+
+/**
+ * Helpers functions.
+ */
+
+const relativeToAbsolute = ( basePath, relativePath ) => {
+	const target = path.join( path.dirname( basePath ), relativePath );
+	if ( path.extname( target ) === '.js' ) {
+		return target;
+	}
+	let targetFile = target + '.js';
+	if ( fs.existsSync( targetFile ) ) {
+		return targetFile;
+	}
+	targetFile = path.join( target, 'index.js' );
+	if ( fs.existsSync( targetFile ) ) {
+		return targetFile;
+	}
+	process.stdout.write( '\nRelative path does not exists.' );
+	process.stdout.write( '\n' );
+	process.stdout.write( `\nBase: ${ basePath }` );
+	process.stdout.write( `\nRelative: ${ relativePath }` );
+	process.stdout.write( '\n\n' );
+	process.exit( 1 );
+};
+
+const getIRFromRelativePath = ( rootDir, basePath ) => ( relativePath ) => {
+	if ( ! relativePath.startsWith( '.' ) ) {
+		return [];
+	}
+	const absolutePath = relativeToAbsolute( basePath, relativePath );
+	const result = processFile( rootDir, absolutePath );
+	return result.ir || undefined;
+};
+
+const processFile = ( rootDir, inputFile ) => {
+	try {
+		const data = fs.readFileSync( inputFile, 'utf8' );
+		currentFileStack.push( inputFile );
+		const relativePath = path.relative( rootDir, inputFile );
+		const result = engine( relativePath, data, getIRFromRelativePath( rootDir, last( currentFileStack ) ) );
+		currentFileStack.pop( inputFile );
```

Comment:
`Array#pop` takes no arguments. It could be safely removed.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop

Reply:
Mind to add this in the `@private` PR?

Reply:
> Mind to add this in the `@private` PR?

I made a small pull request: #15397